File: Leetcode/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        sign = -1 if x < 0 else 1
        x = abs(x)
        while x != 0:
            result = (result * 10) + (x % 10)
            x //= 10
        result *= sign
        return result if (result < 2147483651 and result > -2147483651) else 0

File: LeetCode/ReverseInteger.cs
public class Solution
{
    public int Reverse(int x)
    {
        int result = 0;
        while (x != 0)
        {
            if (result > int.MaxValue / 10 || result < int.MinValue / 10)
            {
                return 0;
            }
            result = (result * 10) + (x % 10);
            x /= 10;
        }
        
        return result;
    }
} 
